tensorboard/plugins/custom_scalar/custom_scalar_demo.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Create sample PR curve summary data.
+
+The logic below logs scalar data and then lays out the custom scalars dashboard.
+"""
+
+from tensorboard import summary as summary_lib
+from tensorboard.plugins.custom_scalar import layout_pb2
+import tensorflow as tf
+
+step = tf.placeholder(tf.float32, shape=[])
+
+with tf.name_scope('loss'):
+  # Specify 2 different loss values, each tagged differently.
+  summary_lib.scalar('foo', tf.pow(0.9, step))
+  summary_lib.scalar('bar', tf.pow(0.85, step + 2))
+
+  # Log metric baz as well as upper and lower bounds for making a margin chart.
+  middle_baz_value = step + 4 * tf.random_uniform([]) - 2
+  summary_lib.scalar('baz', middle_baz_value)
+
+summary_lib.scalar('baz_lower', middle_baz_value - 0.3 - tf.random_uniform([]))
```

Comment:
Ah, I see.  In actual usage it seems like that could get a bit cumbersome, since I'd more naturally think of grouping the bounds under the same prefix, but I see the downside now.  I wonder if adding a negative-match regex to multiline plots could be a decent solution, so that you could have tags like

loss/baz/scalar_summary
loss/baz_margins/lower/scalar_summary
loss/baz_margins/upper/scalar_summary

And then plot loss.* with a negative filter on loss/.*margins.*.  Anyway, doesn't need to be resolved here, just a thought.